dockernet/src/init_chain.sh

```diff
@@ -0,0 +1,178 @@
+echo "$NUM_NODES"
+for (( i=1; i <= $NUM_NODES; i++ )); do
+    # Node names will be of the form: "simd1"
+    node_name="${NODE_PREFIX}${i}"
+    # Moniker is of the form: SIMD_1
+    moniker=$(printf "${NODE_PREFIX}_${i}" | awk '{ print toupper($0) }')
+
+    # Create a state directory for the current node and initialize the chain
+    mkdir -p $STATE/$node_name
+    
+    # If the chains commands are run only from docker, grab the command from the config
+    # Otherwise, if they're run locally, append the home directory
+    cmd="$BINARY --home ${STATE}/$node_name"
+    # Initialize the chain
+    $cmd init $moniker --chain-id $CHAIN_ID --overwrite &> /dev/null
+    chmod -R 777 $STATE/$node_name
+
+    # Update node networking configuration 
+    config_toml="${STATE}/${node_name}/config/config.toml"
+    client_toml="${STATE}/${node_name}/config/client.toml"
+    app_toml="${STATE}/${node_name}/config/app.toml"
+    genesis_json="${STATE}/${node_name}/config/genesis.json"
+
+    sed -i -E "s|cors_allowed_origins = \[\]|cors_allowed_origins = [\"\*\"]|g" $config_toml
+    sed -i -E "s|127.0.0.1|0.0.0.0|g" $config_toml
+    sed -i -E "s|timeout_commit = \"5s\"|timeout_commit = \"${BLOCK_TIME}\"|g" $config_toml
+    sed -i -E "s|prometheus = false|prometheus = true|g" $config_toml
+
+    sed -i -E "s|minimum-gas-prices = \".*\"|minimum-gas-prices = \"0${DENOM}\"|g" $app_toml
+    sed -i -E '/\[api\]/,/^enable = .*$/ s/^enable = .*$/enable = true/' $app_toml
+    sed -i -E 's|unsafe-cors = .*|unsafe-cors = true|g' $app_toml
+    sed -i -E "s|snapshot-interval = 0|snapshot-interval = 300|g" $app_toml
+    sed -i -E 's|localhost|0.0.0.0|g' $app_toml
+
+    sed -i -E "s|chain-id = \"\"|chain-id = \"${CHAIN_ID}\"|g" $client_toml
+    sed -i -E "s|keyring-backend = \"os\"|keyring-backend = \"test\"|g" $client_toml
+    sed -i -E "s|node = \".*\"|node = \"tcp://localhost:$RPC_PORT\"|g" $client_toml
+
+    sed -i -E "s|\"stake\"|\"${DENOM}\"|g" $genesis_json 
+    sed -i -E "s|\"aphoton\"|\"${DENOM}\"|g" $genesis_json # ethermint default
+
+    # add a validator account
+    val_acct="${VAL_PREFIX}${i}"
+    val_mnemonic="${VAL_MNEMONICS[((i-1))]}"
+    echo "$val_mnemonic" | $cmd keys add $val_acct --recover --keyring-backend=test >> $KEYS_LOGS 2>&1
+    val_addr=$($cmd keys show $val_acct --keyring-backend test -a | tr -cd '[:alnum:]._-')
+    # Add this account to the current node
+    $cmd add-genesis-account ${val_addr} ${VAL_TOKENS}${DENOM}
+
+    # Copy over the provider simd validator keys to the provider (in the event
+    # that we are testing ICS)
+    if [[ $CHAIN == "GAIA" && -d $DOCKERNET_HOME/state/${SIMD_NODE_PREFIX}${i} ]]; then
+        simd_config=$DOCKERNET_HOME/state/${SIMD_NODE_PREFIX}${i}/config
+        host_config=$DOCKERNET_HOME/state/${NODE_PREFIX}${i}/config
+        cp ${simd_config}/priv_validator_key.json ${host_config}/priv_validator_key.json
+        cp ${simd_config}/node_key.json ${host_config}/node_key.json
+    fi
```

Comment:
do we need this part?

Reply:
We don't need this, I going to remove this part.